test/gpu/stream_sync.cpp

```diff
@@ -0,0 +1,212 @@
+const std::string compare_numbers = R"__migraphx__(
+#include <hip/hip_runtime.h>
+
+extern "C" {
+__global__ void compare(float* data) 
+{
+    int i = threadIdx.x + blockDim.x * blockIdx.x;
+    if (data[i] != 1337) 
+    {
+        //abort();
+    }
+}
+    
+}
+
+int main() {}
+
+)__migraphx__";
+
+migraphx::src_file make_src_file(const std::string& name, const std::string& content)
+{
+    return {name, std::make_pair(content.data(), content.data() + content.size())};
+}
+
+TEST_CASE(test_stream_sync_compare_kernel)
+{
+    auto binaries = migraphx::gpu::compile_hip_src(
+        {make_src_file("check_stuff.cpp", compare_numbers)}, "", migraphx::gpu::get_device_name());
+    EXPECT(binaries.size() == 1);
+
+    migraphx::gpu::kernel k1{binaries.front(), "compare"};
+
+    auto input =
+        migraphx::fill_argument({migraphx::shape::float_type, {128}}, STREAM_SYNC_TEST_VAL);
+
+    auto ginput = migraphx::gpu::to_gpu(input);
+
+    hipStream_t stream;
+    auto status = hipStreamCreate(&stream);
+    if(status != hipSuccess)
+    {
+        MIGRAPHX_THROW("Failed to get stream");
+    }
+
+    k1.launch(stream, input.get_shape().elements(), 1024)(ginput.cast<std::float_t>());
+
+    status = hipStreamDestroy(stream);
+    if(status != hipSuccess)
+    {
+        MIGRAPHX_THROW("Failed to cleanup stream");
+    }
+
+    auto output = migraphx::gpu::from_gpu(ginput);
+    EXPECT(output == input);
+}
+
+TEST_CASE(test_stream_sync_different_stream)
+{
+    auto binaries = migraphx::gpu::compile_hip_src(
+        {make_src_file("check_stuff.cpp", compare_numbers)}, "", migraphx::gpu::get_device_name());
+    EXPECT(binaries.size() == 1);
+
+    migraphx::gpu::kernel k1{binaries.front(), "compare"};
+
+    // Setup empty GPU memory buffer
+    migraphx::shape io_shape{migraphx::shape::float_type, {128, 128}};
+    auto input  = migraphx::fill_argument(io_shape, 0);
+    auto ginput = migraphx::gpu::to_gpu(input);
+
+    auto output  = migraphx::fill_argument(io_shape, 0);
+    auto goutput = migraphx::gpu::to_gpu(output);
+
+    hipStream_t stream;
+    auto status = hipStreamCreate(&stream);
+    if(status != hipSuccess)
+    {
+        MIGRAPHX_THROW("Failed to get stream");
+    }
+
+    hipStream_t kernel_stream;
+    status = hipStreamCreate(&kernel_stream);
+    if(status != hipSuccess)
+    {
+        MIGRAPHX_THROW("Failed to get stream2");
+    }
+
+    migraphx::program p;
+    auto* mm      = p.get_main_module();
+    auto x        = mm->add_parameter("x", io_shape);
+    auto y        = mm->add_literal({io_shape, {STREAM_SYNC_TEST_VAL + 100}});
+    auto test_val = mm->add_literal({io_shape, {STREAM_SYNC_TEST_VAL}});
+
+    auto mult_out = mm->add_instruction(migraphx::make_op("mul"), x, y);
+    auto add_out  = mm->add_instruction(migraphx::make_op("add"), mult_out, test_val);
+    mm->insert_parameter(add_out, "output", io_shape);
+
+    migraphx::compile_options options;
+    p.compile(migraphx::gpu::target{}, options);
```

Comment:
This can just be `p.compile(migraphx::gpu::target{})` since its just using the default options.

Reply:
Done